### ISSUE TITLE
Consider using in-config defaults instead of in-code defaults

### ISSUE THREAD

**Issue by ktoso** [akka/akka#19948]
AFAIR we manually handle "first try here, then try here" but could instead do (and document for people):

```
# Otherwise you need to create differently named sections containing
# only those settings that shall be different from the defaults
# configured here, importing the defaults like so:
#
#   my-dynamodb-journal = ${dynamodb-journal}
#   my-dynamodb-journal {
#     <settings...>
#   }
```

See here: https://github.com/akka/akka-persistence-dynamodb/blob/master/src/main/resources/reference.conf#L6

**Comment by agolubev** [comment]
@ktoso should it be done for only akka.http?

**Comment by ktoso** [comment]
No, it's not HTTP specific I think, but most offenders are likely to be here (due to nature of multiple nesting etc)

**Comment by agolubev** [comment]
hm, so let me start with http and in case it's clear and straightforward will do the rest.

**Comment by ktoso** [comment]
Cool, thanks a lot :)

**Comment by agolubev** [comment]
I tried to image how it suppose to be. We need some "default" config space (assuming this is what we have in reference.conf's now) and some custom "my", "my1" spaces

Dynamodb plugin has enter point as akka.persistence.journal.plugin, with value of attribute name prefix. So it's easy to define name for space with fallback space

On the other hand for akka http it looks like:
```object ClientConnectionSettingsImpl extends SettingsCompanion[ClientConnectionSettingsImpl]("akka.http.client") {```  
Here I believe it's still possible to have some attribute that will define custom space name with fallback like `akka.http.myclient`

Also for akka in general there is class `Settings` that has a lot of fully qualified names for different setting names like: akka.actor.debug.event-stream. And a lot of other subprojects uses the same full name.  So it would be hard to introduce attribute for global custom space (with fallback). 

To sum up. I can see the only way to introduce custom space for HTTP as it's isolated. Maybe you have some examples or ideas you can share.

**Comment by agolubev** [comment]
Per discussion with @ktoso: to fix this globally we'll need to make big effort on adding prefix to all/subset of settings (as now akka using absolute paths). 

Considering this ticket on hold to get feedback from core team members.

**Comment by jrudolph** [comment]
Ticket moved to http://github.com/akka/akka-http
 Rationale for the move discussed here: akka/akka-meta#27

If you are interested in contributing or tracking this issue, please comment in akka-http instead from now on :-)
